### PR TITLE
refactor: mf-4173 refactor addresses controlling

### PR DESCRIPTION
**Pull request by UncleBill**
https://mask.atlassian.net/browse/MF-4173

### CODE REVIEW

File: packages/web3-hooks/base/src/useHiddenAddressConfig.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      }
+/**
+ *
+ * Get unlisted address by persona pubkey.
+ *
+ * @param personaPubkey The persona identifier in hex string
+ * @param pluginID The plugin id as store key
+ * @returns
+ */
+export function useHiddenAddressConfig(personaPubkey?: string, pluginID?: PluginID) {
```
Comment (UncleBill) [inline packages/web3-hooks/base/src/useHiddenAddressConfig.ts:20]: Recorder the parameters from `[pluginId, pubkey]` to `[pubkey, pluginId]` according to storage fields order.

**Review by guanbinrui — DISMISSED** [review]

**Comment by UncleBill** [comment]
@guanbinrui Just fixed a small problem, PTAL.

File: packages/web3-shared/base/src/specs/index.ts [hunk 2]
```diff
@@ -1360,6 +1363,19 @@ export interface HubState<
     getHub?: (options: Web3HubOptions) => Web3Hub
 }
 
+/** To create a writable storage, you need to pass a signer as parameter */
+declare function CreateNextIDStorage(
```
Comment (guanbinrui) [inline packages/web3-shared/base/src/specs/index.ts:1367]: What about moving them to the NextID provider API?

File: packages/shared/src/UI/components/WalletSettingCard/UI.tsx [hunk 3]
```diff
@@ -6,8 +6,8 @@ import { WalletIcon } from '../WalletIcon/index.js'
 
 const useStyles = makeStyles()((theme) => ({
     root: {
-        boxShadow:
-            theme.palette.mode === 'dark' ? '0px 0px 20px rgba(255, 255, 255, 0.12)' : '0 0 20px rgba(0, 0, 0, 0.05)',
+        // boxShadow:
+        // theme.palette.mode === 'dark' ? '0px 0px 20px rgba(255, 255, 255, 0.12)' : '0 0 20px rgba(0, 0, 0, 0.05)',
```
Comment (guanbinrui) [inline packages/shared/src/UI/components/WalletSettingCard/UI.tsx]:
```suggestion
```

File: packages/web3-shared/base/src/specs/index.ts (hunk 2 above)
Comment (UncleBill) [reply packages/web3-shared/base/src/specs/index.ts:1367]: This web3-shared package should not depend on the NextID provider.

File: packages/web3-shared/base/src/specs/index.ts [hunk 4]
```diff
@@ -1360,6 +1363,19 @@ export interface HubState<
...
+    proofIdentity: string,
+    platform: NextIDPlatform,
+    signerOrPublicKey: string,
```
Comment (UncleBill) [inline packages/web3-shared/base/src/specs/index.ts]:
```suggestion
    publicKey: string,
```

File: packages/web3-shared/base/src/specs/index.ts [hunk 5]
```diff
@@ -1360,6 +1363,19 @@ export interface HubState<
...
+): ReadonlyStorage
+
+declare function CreateNextIDStorage(
+    proofIdentity: string,
+    platform: NextIDPlatform,
+    signerOrPublicKey: ECKeyIdentifier,
```
Comment (UncleBill) [inline packages/web3-shared/base/src/specs/index.ts]:
```suggestion
    signer: ECKeyIdentifier,
```

File: packages/shared-base/src/data/queryClient.ts [hunk 6]
```diff
@@ -5,7 +5,6 @@ export const queryClient = new QueryClient({
         queries: {
             staleTime: 20_000,
             cacheTime: 300_000,
-            refetchOnWindowFocus: true,
```
Comment (Jack-Works) [inline packages/shared-base/src/data/queryClient.ts:8]: you can only disable it in the development mode.

Comment (UncleBill) [reply packages/shared-base/src/data/queryClient.ts:8]: This option is true by default, so I just remove it.

**Review by guanbinrui — APPROVED** [review]

File: packages/shared-base/src/data/queryClient.ts (hunk 6 above)
Comment (Jack-Works) [reply packages/shared-base/src/data/queryClient.ts:8]: this option improves the user experience

**Pull request merged**